### PR TITLE
fix: Add TypeMeta to observability-proxy-credentials secret

### LINKED ISSUE #1592: Object 'Kind' is missing in observability-proxy-credentials
I'm seeing output like the following in the KFM logs with OSD 4.12:
```
E0222 12:01:49.594155       1 logger.go:210] failed to reconcile provisioned cluster xxxx: failed to reconcile cluster xxxx SyncSet: failed to apply resources for cluster xxxx: KAFKAS-MGMT-9: failed to apply resources ext-managedservice-cluster-mgr
 caused by: Object 'Kind' is missing in '{"metadata":{"name":"observability-proxy-credentials","namespace":"managed-application-services-observability","creationTimestamp":null},"stringData":{"client_id":"","client_secret":"","issuer_url":"https://sso-keycloak.apps.xxx-kafka.j1j7.s1.devshift.org/auth/realms/rhoas"}}': failed to apply resources for cluster xxxx: KAFKAS-MGMT-9: failed to apply resources ext-managedservice-cluster-mgr
 caused by: Object 'Kind' is missing in '{"metadata":{"name":"observability-proxy-credentials","namespace":"managed-application-services-observability","creationTimestamp":null},"stringData":{"client_id":"","client_secret":"","issuer_url":"https://sso-keycloak.apps.xxx-kafka.j1j7.s1.devshift.org/auth/realms/rhoas"}}'
```
Adding `TypeMeta` to the secret named in the error fixes the issue. I'll open a PR to resolve this.

### PR DESCRIPTION
**Pull request by MikeEdgar**
## Description
OpenShift 4.12 returns error creating secret `observability-proxy-credentials` due to missing `kind` property. Fixes #1592

## Verification Steps

1. Install using kas-installer using version 4.12 OSD and `main` branch. The installer will be stuck waiting for the creation of the fleetshard managed-agent. Checking the KFM logs, the error listed in #1592 will be seen.
2. Re-run kas-installer using the changes in this PR by placing the following configs in kas-installer.env:
    ```
    KAS_FLEET_MANAGER_GIT_URL='git@github.com:MikeEdgar/kas-fleet-manager.git'
    KAS_FLEET_MANAGER_GIT_REF='add-secret-typemeta'
    KAS_FLEET_MANAGER_IMAGE_BUILD='true'
    KAS_FLEET_MANAGER_IMAGE_REPOSITORY="${USER}/kas-fleet-manager"
    KAS_FLEET_MANAGER_IMAGE_TAG='latest'
    ```
3. Installation will complete successfully

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

File: internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go [hunk 1]
```diff
@@ -881,6 +881,10 @@ func (c *ClusterManager) buildObservatoriumDexSecretResource() *k8sCoreV1.Secret
 
 func (c *ClusterManager) buildObservabilityRemoteWriteServiceAccountCredential(cluster *api.Cluster) *k8sCoreV1.Secret {
 	return &k8sCoreV1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: metav1.SchemeGroupVersion.Version,
```
Comment (miguelsorianod) [inline internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go:885]: I am not sure this is correct.
The APIVersion of a Secret is `v1` (or `core/v1`), not `meta.k8s.io/v1` which is what is being referenced there

Comment (miguelsorianod) [reply internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go:885]: This should be set to `k8sCoreV1.SchemeGroupVersion.String()` instead

**Review by miguelsorianod — CHANGES_REQUESTED** [review]
See feedback

**Comment by miguelsorianod** [comment]
This also makes me think that changes in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1560 where not properly tested. Please let's be more careful with the verifications

**Comment by miguelsorianod** [comment]
> OpenShift 4.12 returns error creating secret observability-proxy-credentials due to missing kind property. Fixes https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/issues/1592

Out of curiosity, did it also fail in other OCP versions?

File: internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go (hunk 1 above)
Comment (MikeEdgar) [reply internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go:885]: This seems to be the `APIVersion` used for secrets throughout the application.

**Comment by MikeEdgar** [comment]
> > OpenShift 4.12 returns error creating secret observability-proxy-credentials due to missing kind property. Fixes #1592
> 
> Out of curiosity, did it also fail in other OCP versions?

@miguelsorianod I did not try it with 4.10 or 4.11.

File: internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go (hunk 1 above)
Comment (miguelsorianod) [reply internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go:885]: See K8s Secret API doc: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#secret-v1-core

Comment (MikeEdgar) [reply internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go:885]: For this change, would it be fine to reference the correct group version and deal with the others in a separate issue/PR?

Comment (miguelsorianod) [reply internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go:885]: > For this change, would it be fine to reference the correct group version and deal with the others in a separate issue/PR?

Yes. I will take care of fixing it for the others

**Comment by machi1990** [comment]
Thanks @MikeEdgar  @miguelsorianod

File: internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go [hunk 2]
```diff
@@ -881,6 +881,10 @@ func (c *ClusterManager) buildObservatoriumDexSecretResource() *k8sCoreV1.Secret
 
 func (c *ClusterManager) buildObservabilityRemoteWriteServiceAccountCredential(cluster *api.Cluster) *k8sCoreV1.Secret {
 	return &k8sCoreV1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: k8sCoreV1.SchemeGroupVersion.Version,
```
Comment (miguelsorianod) [inline internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go:885]: Unfortunately this is not fully correct still.
`APIVersion` contains the API Group and the API Version, not only the version. It is a confusing name.
In this case `k8sCoreV1.SchemeGroupVersion.String()` should be used instead.
See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#resources

**Review by miguelsorianod — CHANGES_REQUESTED** [review]
.

**Review by miguelsorianod — APPROVED** [review]
Looks good to me :+1: .
Let's wait until tests pass and if there are no merge conflicts we can merge

**Pull request merged**